### PR TITLE
Implement pbr-material for fantasia3d-texture pipeline

### PR DESCRIPTION
**Pull request by bennyguo**
This PR is from [this PR](https://github.com/threestudio-project/threestudio/pull/173). I've made the following changes:

- The MLP network is removed from the material, i.e., the material simply explains what the features given by the geometry should be (albedo, roughness, ...) without doing additional network evaluations. Actually I'm not really sure whether we should put MLP networks in the material, but this is consistent with other existing material implementations (like the `diffuse-with-point-light-material`).
- `train_normal_map` in the `pbr-material` is renamed to `use_bump`. I think it's better to use the word `bump` to represent normal perturbation, so that it's distinguished from the actual geometric normal.
- Add a `requires_tangent` attribute to `BaseMaterial`, so that the rasterizer does not compute tangent space (which requires time-consuming UV unwrapping if the mesh contains too many faces) if the material does not require tangent.
- Use `solid-color-background` and a learning rate of 0.01 for Fantasia3D texture stage training. This (1) is aligned with the official implementation; (2) speeds up convergence; (3) and I empirically found that using learned background sometimes leads to degraded color. Let me know if you find this strategy to be problematic in some cases.

### CODE REVIEW

**Comment by bennyguo** [comment]
An example of degraded color caused by learned background:

learned background:
image

solid white background:
image

**Comment by ashawkey** [comment]
@bennyguo Great! I think the modifications solve my problems. 
About the background, the original implementation uses solid black color. In experiments I found part of the object tends to be black too and cannot distinguish from background, so I changed it to the trainable background. 
For the degradation problem, I guess background `random_aug` should be able to help, but that's too different from the paper in reproduction meaning.
To balance between performance and aligning with the paper, I think the current solid-white is better.

**Comment by bennyguo** [comment]
I think you're right. Theoretically, white background could also suffer from such a problem. I suspected that the degradation may result from the large learning rate I'm using, i.e., the background is learned too fast. But things did not get better even if  I change the background learning rate to 0.001. So I'll leave it this way for now. Thanks again for the contribution! @ashawkey

**Review by thuliu-yt16 — APPROVED** [review]
It all looks good. I just left a few comments on the normal computation in nvdiff-rasterizer. And it seems to need imageio>=2.28.0 to work well with reading the hdr images.

**Comment by bennyguo** [comment]
@ashawkey We should probably consider adding the `imageio>=2.28.0` constraint to the `envlight` package.

**Comment by ashawkey** [comment]
Thanks for spotting that, I have updated the dependency!

**Comment by ARuiChen** [comment]
@ashawkey 
Actually, I used random backgrounds referring to CLIPMesh, including checkboard, blurred Gaussian, and so on. But I found it easily bake the background color to the diffuse map in some cases, so I discard it.

**Comment by ashawkey** [comment]
Yes, it is a common problem in all dreamfusion-based methods.
